### ISSUE TITLE
Collapse `Number Slider` and `Integer Slider` Into One Node

### ISSUE THREAD

**Issue by ikeough** [DynamoDS/Dynamo#7778]
We should collapse `Number Slider` and `Integer Slider` into one node called `Number Slider`, and we should add a `Precision` field to the settings available on the node to allow the user to specify `Double` or `Integer`.

A migration would be applied that takes existing `Integer Slider` nodes and replaces them with `Number Slider` nodes with the `Precision` set to `Integer`.

This issue is related to #7747. Collapsing the two nodes into one allows us to reduce the number of base node types in the graph schema.

**Comment by gregmarr** [comment]
@ikeough Did you mean to say `NumberType` instead of `Precision` above?  Should this also be added to the `Input.Number` node?

**Comment by ikeough** [comment]
It doesn't need to be added to the `Input.Number` node because, I believe, the VM treats everything as doubles currently.

**Comment by gregmarr** [comment]
Yes, it would be mostly a UI thing, not a change to the data type, just like with the slider.

**Comment by wynged** [comment]
Can it be a more generic precision input rather than int/double?  what if I
want the precision to be 0.5?

**Comment by gregmarr** [comment]
That would be the existing "Step" field.

**Comment by wynged** [comment]
oh, right.... so why is the precision toggle important?  if the number
slider produces integers, can it simply cast them as integers rather having
the user to manually change this?

**Comment by gregmarr** [comment]
It specifies the user's intent to only produce integer numbers.  That can be important for downstream operations.  It can also help by restricting the min, max, and step values to integers, so that you don't accidentally put in a non-integer value.

**Comment by Racel** [comment]
This was proposed as part of the library re-categorization effort. I support this idea. So, @ikeough @gregmarr, are you suggesting that we add an additional property to the "Number Slider" node where the user selects integer or double? Or will it be enough to specify "1" vs "1.0" in the step value? What will the default be set to? double?

**Comment by gregmarr** [comment]
I'd say an additional property defaulting to double.

**Comment by smangarole** [comment]
We will be working on this and will be adding a task in the backlog.

**Comment by Thomas84** [comment]
In my opinion the number slider needs also an improvement: The maximum is always used as highest number, regardless of the step value (someone told me it comes from WPF). It should have the same behavior as the Range node: 
![image](https://user-images.githubusercontent.com/5624034/52058900-b8df9e00-2568-11e9-858d-d97e633e254c.png)
